Setup SAMADhi

OlivierBondu opened the pull request:
Automatic script putting the sample of the framework output in the database

@delaere : currently I put https://github.com/cp3-llbb/GridIn/compare/cp3-llbb:master...OlivierBondu:setup_SAMADhi?expand=1#diff-b975cf591369e83050619b4cba81fad1R183, but due to https://github.com/delaere/SAMADhi/blob/292dd4fb8ec09ed3547cf48cecd2385172e553fd/scripts/add_sample.py#L108-L109 it is generating a lot of output.... Do you have an idea of a work-around without breaking the functionality ?

delaere commented:
@OlivierBondu : I see two possible ways out. Either we add a "batch mode" to SAMADhi, that would disable all prompts and assume the defaults, or use directly the python commands instead of calling the script (~10 lines from the script is enough, a bit more with basic checks). add_sample.py was not designed for use in other scripts.

OlivierBondu commented:
Updated: I removed the dependance on `add_sample.py` by copy-pasting the essential functionality into this script

(aka: ready to be merged as far as I am concerned)

blinkseb commented on scripts/runPostCrab.py:
```diff
@@ -0,0 +1,219 @@
+        print "ERROR: the content of the path and the list of crab outputs are different, abort now!"
+        print "localfiles (what's here locally)= ", localfiles
+        print "outputfiles (what crab is saying)= ", output_files['lfn']
+        raise AssertionError("CRAB3 and local list of files do not match")
```
Why do you do that? You can imagine some cases where you have duplicate files into your output folder, but you don't care, since you use the list of output files given by crab instead of listing the files in the output directory.

IMO this is not correct.

OlivierBondu replied:
this is due to the [current structure of a sample in the DB](https://github.com/delaere/SAMADhi/blob/af8ed5b552bf7be6e0eee48857251d44f6632256/scripts/SAMADhi.py#L69-L90): it stores the path itself, not the list of files. If we want/need a list of files, we probably need a container for it...

But as far as I have experienced, there are no duplicate entries in the crab output dir, possible duplicates are stored in a subdir (plus, I don't see how there could be duplicates given the output file name of a given job has no 'unique hash' as it was the case in crab2)

blinkseb replied:
Ah ok ... Well I guess it'd be better to store a list of files instead of a directory into the database but in the mean time this'll fine.

I was mostly about my crab2 experience. It's good to see they seem to have changed how failed jobs are treated :smile:

OlivierBondu replied:
Yes, I don't expect this to be final, but rather a working placeholder :smile: 

see the bunch of FIXME existing in the script, in particular I expect [this line](https://github.com/cp3-llbb/GridIn/pull/6/files#diff-b975cf591369e83050619b4cba81fad1R158) to break for the first big sample we'll have to process....

(plus I am sure we'll have crab3 surprises in the same way we had crab2 surprises like the one you mentioned...)
